### PR TITLE
{Release} Upgrade to Azure CLI 2.21.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,132 @@
...
 
+2.21.0
+++++++
+
+**ACR**
+
+* Output a trace in "az acr login" for self-diagnosing potential docker command latency (#17115)
+* Fix a bug when run check-health behind corporate proxy (#17177)
+* Support anonymous pull (#17006)
+* Fix#16700: Use "exists" api to check storage blob existence (#17299)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:14]:
```suggestion
* Fix #16700: Use "exists" api to check storage blob existence (#17299)
```

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,132 @@
 Release History
 ===============
 
+2.21.0
+++++++
+
+**ACR**
+
+* Output a trace in "az acr login" for self-diagnosing potential docker command latency (#17115)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:11]:
```suggestion
* Output a trace in `az acr login` for self-diagnosing potential docker command latency (#17115)
```

File: src/azure-cli/HISTORY.rst (hunk 1 above)
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:14]:
```suggestion
* Fix #16700: Use "exists" api to check storage blob existence (#17299)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,132 @@
...
+* Fix a bug when run check-health behind corporate proxy (#17177)
+* Support anonymous pull (#17006)
+* Fix#16700: Use "exists" api to check storage blob existence (#17299)
+
+**AKS**
+
+* Add --no-uptime-sla (#17192)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:18]:
```suggestion
* Add `--no-uptime-sla` (#17192)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,132 @@
...
+**APIM**
+
+* [BREAKING CHANGE] `apim backup`: `--storage-account-container` not support multi-value. (#17315)
+* [BREAKING CHANGE] `apim restore`: `--storage-account-container` not support multi-value. (#17315)
+
+**App Service**
+
+* [BREAKING CHANGE] Fixes #16087:az webapp config ssl create  setting --name parameter as required. (#17079)
+* : fix #17053- `az webapp show` returning null values for SiteConfig properties (#17054)
+* : fix #17207 - for az webapp log config 'level' always defaults to verbose (#17259)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:31]:
```suggestion
* : Fix #17053: `az webapp show` returning null values for SiteConfig properties (#17054)
* : Fix #17207: for az webapp log config 'level' always defaults to verbose (#17259)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,132 @@
...
+
+**Compute**
+
+* Fix a Windows vm user update bug (#17257)
+* Fix #16585: az vmss deallocate --instance-ids failed (#17274)
+* vm create: New parameter --platform-fault-domain in FLEX VMSS mode (#16409)
+* vm create: --patch-mode for Linux VM (#16409)
+* `az ssh vm`: Automatically launch browser when getting certificate fails (#17093)
+* vm create: New parameter --count (#17217)
+* vm create: Trusted Launch (#17354)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:59]:
```suggestion
* `az vm create`: New parameter --platform-fault-domain in FLEX VMSS mode (#16409)
* `az vm create`: --patch-mode for Linux VM (#16409)
* `az ssh vm`: Automatically launch browser when getting certificate fails (#17093)
* `az vm create`: New parameter --count (#17217)
* `az vm create`: Trusted Launch (#17354)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,132 @@
...
+* : fix #17053- `az webapp show` returning null values for SiteConfig properties (#17054)
+* : fix #17207 - for az webapp log config 'level' always defaults to verbose (#17259)
+
+**ARM**
+
+* `az bicep build`: fix an issue where build warnings are not shown (#17180)
+
+**Backup**
+
+* Added id_part for sub-resource names (#17165)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:39]:
```suggestion
* Add `id_part` for sub-resource names to fix `--ids` (#17165)
```

File: src/azure-cli/HISTORY.rst (hunk 5 above)
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:59]:
```suggestion
* `az vm create`: Trusted Launch (#17354)
* `az vm open-port`: Accept list of ports (#17255)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,132 @@
...
 ===============
 
+2.21.0
+++++++
+
+**ACR**
+
+* Output a trace in `az acr login` for self-diagnosing potential docker command latency (#17115)
+* Fix a bug when run check-health behind corporate proxy (#17177)
+* Support anonymous pull (#17006)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:13]:
```suggestion
* `acr update`: Support anonymous pull (#17006)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,132 @@
...
+
+* Allow configuring theme (#17073)
+
+**Upgrade**
+
+* Fix 16952: Fix ImportError after upgrade (#17314)
+
+**VM**
+
+* Add #16037: az vm open-port accepts list of ports (#17255)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:130]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,132 @@
...
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst]:
```suggestion
* Fix 16952: Fix ImportError after upgrade (#17314)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,128 @@
...
+
+**Compute**
+
+* Fix a Windows vm user update bug (#17257)
+* Fix #16585: az vmss deallocate --instance-ids failed (#17274)
+* `az vm create`: New parameter --platform-fault-domain in FLEX VMSS mode (#16409)
+* `az vm create`: --patch-mode for Linux VM (#16409)
+* `az ssh vm`: Automatically launch browser when getting certificate fails (#17093)
+* `az vm create`: New parameter --count (#17217)
+* `az vm create`: Trusted Launch (#17354)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:59]:
```suggestion
* `az vm create`: Trusted Launch (#17354)
* Fix #16037: az vm open-port accepts list of ports (#17255)
```

File: src/azure-cli/HISTORY.rst [hunk 11]
```diff
@@ -3,6 +3,128 @@
...
+* Add support for node public IP prefix ID (#17138)
+
+**APIM**
+
+* [BREAKING CHANGE] `apim backup`: `--storage-account-container` not support multi-value. (#17315)
+* [BREAKING CHANGE] `apim restore`: `--storage-account-container` not support multi-value. (#17315)
+
+**App Service**
+
+* [BREAKING CHANGE] Fixes #16087:az webapp config ssl create  setting --name parameter as required. (#17079)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:29]:
```suggestion
* [BREAKING CHANGE] Fixes #16087: `az webapp config ssl create setting` `--name` parameter as required. (#17079)
```

File: src/azure-cli/HISTORY.rst [hunk 12]
```diff
@@ -3,6 +3,128 @@
...
+* Fix #16917: az storage account generate-sas fails if a connection string is provided (#17200)
+* Fix #16979: `az storage container create` fails when providing storage container metadata (#17202)
+
+**Style**
+
+* Allow configuring theme (#17073)
+
+**Upgrade**
+
+* Fix 16952: Fix ImportError after upgrade (#17314)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:126]:
```suggestion
* Fix #16952: Fix ImportError after upgrade (#17314)
```

File: src/azure-cli/HISTORY.rst [hunk 13]
```diff
@@ -3,6 +3,128 @@
...
+
+* `az afd rule create`: Fix '--help' message (#17282)
+
+**Compute**
+
+* Fix a Windows vm user update bug (#17257)
+* Fix #16585: az vmss deallocate --instance-ids failed (#17274)
+* `az vm create`: New parameter --platform-fault-domain in FLEX VMSS mode (#16409)
+* `az vm create`: --patch-mode for Linux VM (#16409)
+* `az ssh vm`: Automatically launch browser when getting certificate fails (#17093)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:57]: It is an extension

File: src/azure-cli/HISTORY.rst [hunk 14]
```diff
@@ -3,6 +3,128 @@
...
+**APIM**
+
+* [BREAKING CHANGE] `apim backup`: `--storage-account-container` not support multi-value. (#17315)
+* [BREAKING CHANGE] `apim restore`: `--storage-account-container` not support multi-value. (#17315)
+
+**App Service**
+
+* [BREAKING CHANGE] Fixes #16087:az webapp config ssl create  setting --name parameter as required. (#17079)
+* Fix #17053: `az webapp show` returning null values for SiteConfig properties (#17054)
+* Fix #17207: for az webapp log config 'level' always defaults to verbose (#17259)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:31]:
```suggestion
* Fix #17207: `az webapp log config` `level` always defaults to verbose (#17259)
```

File: src/azure-cli/HISTORY.rst [hunk 15]
```diff
@@ -3,6 +3,128 @@
...
+* Add Server Trust Group commands (#17275)
+
+**Storage**
+
+* Fix #16917: az storage account generate-sas fails if a connection string is provided (#17200)
+* Fix #16979: `az storage container create` fails when providing storage container metadata (#17202)
+
+**Style**
+
+* Allow configuring theme (#17073)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:122]:
```
> az style -h
'style' is misspelled or not recognized by the system.
```

File: src/azure-cli/HISTORY.rst [hunk 16]
```diff
@@ -3,6 +3,128 @@
...
+**Network**
+
+* `az network public-ip prefix  create`: Support `--zone 1 2 3` (#17279)
+* `az network lb  frontend-ip create`: Support `--zone 1 2 3` (#17279)
+* Bump version from '2020-08-01' to '2020-11-01' (#17290)
+* `az network lb address-pool` : Support subnet when creating or updating an IP-based backend pool of a load balancer. (#17336)
+
+**Packaging**
+
+* Add azure-identity dependency (#17277)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:90]: Move to core?

File: src/azure-cli/HISTORY.rst [hunk 17]
```diff
@@ -3,6 +3,128 @@
...
+* Fix #17053: `az webapp show` returning null values for SiteConfig properties (#17054)
+* Fix #17207: for az webapp log config 'level' always defaults to verbose (#17259)
+
+**ARM**
+
+* `az bicep build`: fix an issue where build warnings are not shown (#17180)
+
+**Backup**
+
+* Add `id_part` for sub-resource names to fix `--ids` (#17165)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:39]:
```suggestion
* `backup item show`: Add `id_part` for sub-resource names to fix `--ids` (#17165)
```

File: src/azure-cli/HISTORY.rst [hunk 18]
```diff
@@ -3,6 +3,128 @@
...
+* Fix #17094: Created separate test suite for CRR tests (#17183)
+* `az backup protection check-vm`: Add vm and resource-group as optional params (#16974)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:41]:
```suggestion
* `az backup protection check-vm`: Add `--vm` and `--resource-group` as optional params (#16974)
```

File: src/azure-cli/HISTORY.rst [hunk 19]
```diff
@@ -3,6 +3,128 @@
...
+
+**Cache**
+
+* GA `az cache` (#17264)
+
+**CDN**
+
+* `az afd rule create`: Fix '--help' message (#17282)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:49]:
```suggestion
* `az afd rule create`: Fix `--help` message (#17282)
```

File: src/azure-cli/HISTORY.rst [hunk 20]
```diff
@@ -3,6 +3,128 @@
...
+
+**Compute**
+
+* Fix a Windows vm user update bug (#17257)
+* Fix #16585: az vmss deallocate --instance-ids failed (#17274)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:54]:
```suggestion
* Fix #16585: `az vmss deallocate` `--instance-ids` failed (#17274)
```

File: src/azure-cli/HISTORY.rst [hunk 21]
```diff
@@ -3,6 +3,128 @@
...
+* `az vm create`: New parameter --platform-fault-domain in FLEX VMSS mode (#16409)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:55]:
```suggestion
* `az vm create`: New parameter `--platform-fault-domain` in FLEX VMSS mode (#16409)
```

File: src/azure-cli/HISTORY.rst (hunk 10 above)
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:59]:
```suggestion
* `az vm create`: `--patch-mode` for Linux VM (#16409)
* `az ssh vm`: Automatically launch browser when getting certificate fails (#17093)
* `az vm create`: New parameter `--count` (#17217)
* `az vm create`: Trusted Launch (#17354)
```

File: src/azure-cli/HISTORY.rst [hunk 22]
```diff
@@ -3,6 +3,128 @@
...
+
+* `az keyvault role definition list`: Support `--custom-role-only` to list only custom role definitions (#17119)
+* Support keyvault custom role definition (#17109)
+* Add `--no-wait` for command `az keyvault security-domain download` and `--target-operation` for command `az keyvault security-domain wait` (#17263)
+
+**NetAppFiles**
+
+* `az netappfiles account backup show`: Operation added. (#17173)
+* `az netappfiles account backup delete`: Operation added. (#17173)
+* `az netappfiles account ad add`: Parameter ldap-over-tls added. (#17173)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:75]:
```suggestion
* `az netappfiles account ad add`: Parameter `--ldap-over-tls` added. (#17173)
```

File: src/azure-cli/HISTORY.rst [hunk 23]
```diff
@@ -3,6 +3,128 @@
...
+* `az netappfiles account create`: Parameter encryption added. (#17173)
+* `az netappfiles account update`: Parameter encryption added. (#17173)
+* `az netappfiles volume create`: Parameter encryption-key-source added. (#17173)
+* `az netappfiles volume create`: Default export policy removed for nfsv4.1 and optional parameters added for setting up an export policy for nfsv4.1: rule_index, unix_read_only, unix_read_write, cifs, allowed_clients (#17173)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:79]:
```suggestion
* `az netappfiles account create`: Parameter `--encryption` added. (#17173)
* `az netappfiles account update`: Parameter `--encryption` added. (#17173)
* `az netappfiles volume create`: Parameter `--encryption-key-source` added. (#17173)
* `az netappfiles volume create`: Default export policy removed for nfsv4.1 and optional parameters added for setting up an export policy for nfsv4.1: rule_index, unix_read_only, unix_read_write, cifs, allowed_clients (#17173)
```

File: src/azure-cli/HISTORY.rst [hunk 24]
```diff
@@ -3,6 +3,128 @@
...
+
+* `az sf certificate`: deprecate cluster cert commands. (#17190)
+
+**SQL**
+
+* Add Server Trust Group commands (#17275)
+
+**Storage**
+
+* Fix #16917: az storage account generate-sas fails if a connection string is provided (#17200)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:117]:
```suggestion
* Fix #16917: `az storage account generate-sas` fails if a connection string is provided (#17200)
```

File: src/azure-cli-core/HISTORY.rst [hunk 25]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+2.21.0
+++++++
+* Unify ADAL and MSAL error handler (#17072)
+* Add support for specifying minimum required version of an extension to work with the current core version (#16751)
+
```
Comment (qwordy) [inline src/azure-cli-core/HISTORY.rst:10]:
```suggestion
* Allow configuring theme (#17073)

```

File: src/azure-cli/HISTORY.rst [hunk 26]
```diff
@@ -3,6 +3,128 @@
 Release History
 ===============
 
+2.21.0
+++++++
+
+**ACR**
+
+* Output a trace in `az acr login` for self-diagnosing potential docker command latency (#17115)
+* Fix a bug when run check-health behind corporate proxy (#17177)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:12]:
```suggestion
* Fix #17172: When run check-health behind corporate proxy (#17177)
```

File: src/azure-cli/HISTORY.rst (hunk 15 above)
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:122]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 27]
```diff
@@ -3,6 +3,124 @@
...
+**Network**
+
+* `az network public-ip prefix  create`: Support `--zone 1 2 3` (#17279)
+* `az network lb  frontend-ip create`: Support `--zone 1 2 3` (#17279)
+* Bump version from '2020-08-01' to '2020-11-01' (#17290)
+* `az network lb address-pool` : Support subnet when creating or updating an IP-based backend pool of a load balancer. (#17336)
+
+**Packaging**
+
+* Add azure-identity dependency (#17277)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:90]:
```suggestion
```

File: src/azure-cli/HISTORY.rst (hunk 16 above)
Comment (jiasli) [reply src/azure-cli/HISTORY.rst:91]: `azure-identity` is required by RDBMS, so belongs to `azure-cli`, not `azure-cli-core`.

File: src/azure-cli-core/HISTORY.rst [hunk 28]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+2.21.0
+++++++
+* Unify ADAL and MSAL error handler (#17072)
+* Add support for specifying minimum required version of an extension to work with the current core version (#16751)
+* Allow configuring theme (#17073)
```
Comment (kairu-ms) [inline src/azure-cli-core/HISTORY.rst:10]:
```suggestion
```

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 29]
```diff
@@ -3,6 +3,126 @@
...
+* Fix #16979: `az storage container create` fails when providing storage container metadata (#17202)
+
+**Upgrade**
+
+* Fix #16952: Fix ImportError after upgrade (#17314)
+
+**Misc.**
+
+* Allow configuring theme (#17073)
+
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:124]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 30]
```diff
@@ -3,6 +3,126 @@
...
+
 2.20.0
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst]:
```suggestion
* Allow configuring theme (#17073)

2.20.0
```

File: src/azure-cli/HISTORY.rst [hunk 31]
```diff
@@ -3,6 +3,125 @@
...
+* Add support for node public IP prefix ID (#17138)
+
+**APIM**
+
+* [BREAKING CHANGE] `apim backup`: `--storage-account-container` not support multi-value. (#17315)
+* [BREAKING CHANGE] `apim restore`: `--storage-account-container` not support multi-value. (#17315)
+
+**App Service**
+
+* [BREAKING CHANGE] Fixes #16087: `az webapp config ssl create`: set `--name` parameter as required. (#17079)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:29]:
```suggestion
* [BREAKING CHANGE] Fix #16087: `az webapp config ssl create`: set `--name` parameter as required. (#17079)
```

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**